klippa/src/parsing_util.rs

```diff
@@ -67,33 +69,65 @@ pub fn parse_unicodes(unicode_str: &str) -> Result<IntSet<u32>, SubsetError> {
 
 //parse input drop_tables string, which is a comma/whitespace-separated list of tables that will be dropped
 pub fn parse_drop_tables(input_str: &str) -> Result<IntSet<Tag>, SubsetError> {
-    let mut result = IntSet::empty();
+    if input_str.trim() == "*" {
+        return Ok(IntSet::<Tag>::all());
+    }
 
-    if input_str == "*" {
-        let out = IntSet::<Tag>::all();
-        return Ok(out);
+    input_str
+        .split(|c| c == ',' || c == ' ')
+        .filter(|raw| !raw.is_empty())
+        .map(|raw| {
+            Tag::new_checked(raw.as_bytes()).map_err(|_| SubsetError::InvalidTag(raw.to_owned()))
+        })
+        .collect()
+}
+
+//parse input name_IDs string, which is a comma/whitespace-separated list of nameIDs that will be retained
+pub fn parse_name_ids(input_str: &str) -> Result<IntSet<NameId>, SubsetError> {
+    if input_str.trim() == "*" {
+        return Ok(IntSet::<NameId>::all());
     }
 
-    for tag in input_str.split(|c| c == ',' || c == ' ') {
-        if tag.is_empty() {
-            continue;
-        }
-        let Ok(table_tag) = Tag::new_checked(tag.as_bytes()) else {
-            return Err(SubsetError::InvalidTag(tag.to_owned()));
-        };
-        result.insert(table_tag);
+    input_str
+        .split(|c| c == ',' || c == ' ')
+        .filter(|raw| !raw.is_empty())
+        .map(|raw| {
+            raw.parse::<u16>()
+                .map(NameId::from)
+                .map_err(|_| SubsetError::InvalidId(raw.to_owned()))
+        })
+        .collect()
+}
+
+//parse input name_languages string, which is a comma/whitespace-separated list of langIDs that will be retained
+pub fn parse_name_languages(input_str: &str) -> Result<IntSet<u16>, SubsetError> {
+    if input_str.trim() == "*" {
+        return Ok(IntSet::<u16>::all());
     }
-    Ok(result)
+
+    input_str
+        .split(|c| c == ',' || c == ' ')
+        .filter(|raw| !raw.is_empty())
+        .map(|raw| {
+            raw.parse::<u16>()
+                .map_err(|_| SubsetError::InvalidId(raw.to_owned()))
+        })
+        .collect()
 }
```

Comment:
I think you can shorten this via a helper function. I tried this locally andthe tests seemed to pass:

```rust
/// Parse a comma or whitespace list of things
fn parse_list<T: Domain>(input_str: &str, parse_one: fn(&str) -> Result<T, SubsetError>) -> Result<IntSet<T>, SubsetError> {
    if input_str.trim() == "*" {
        return Ok(IntSet::all());
    }
    input_str.split(|c| c == ',' || c == ' ')
        .filter(|raw| !raw.is_empty())
        .map(parse_one)
        .collect()
}

//parse input drop_tables string, which is a comma/whitespace-separated list of tables that will be dropped
pub fn parse_drop_tables(input_str: &str) -> Result<IntSet<Tag>, SubsetError> {    
    parse_list(input_str, |raw| Tag::new_checked(raw.as_bytes()).map_err(|_| SubsetError::InvalidTag(raw.to_owned())))
}

//parse input name_IDs string, which is a comma/whitespace-separated list of nameIDs that will be retained
pub fn parse_name_ids(input_str: &str) -> Result<IntSet<NameId>, SubsetError> {
    parse_list(input_str,|raw| raw.parse::<u16>().map(NameId::from).map_err(|_| SubsetError::InvalidId(raw.to_owned())))
}

//parse input name_languages string, which is a comma/whitespace-separated list of langIDs that will be retained
pub fn parse_name_languages(input_str: &str) -> Result<IntSet<u16>, SubsetError> {    
    parse_list(input_str, |raw| raw.parse::<u16>().map_err(|_| SubsetError::InvalidId(raw.to_owned())))
}
```

Reply:
Also, you can just use `input_str.split(&[',', ' '])` to split on any character in a slice though the closure works too.

Reply:
New learning, the code looks much better now. Thank you!